torchci/lib/bot/retryBot.ts

```diff
@@ -55,14 +54,39 @@ function retryBot(app: Probot): void {
       return;
     }
     const shouldRetry = failedJobs.filter((job) => {
-      // always rerun lint
-      if (workflowName === "lint") {
-        return true;
-      }
       // if the job was cancelled, it was probably an infra error, so rerun
       if (job.conclusion === "cancelled") {
         return true;
       }
+
+      // for builds, rerun if it didn't fail on the actual build step
```

Comment:
The same logic applies for `build`, `lint`, and `test`, so may be it's easier to put them into some data structure?

Reply:
Extracted common logic to a helper function. Still wanted to leave step navigation flexible though.